#import <Foundation/Foundation.h>
#import "Statistics.pb.h"
/*
 - 事件统计打点定位，复制粘贴以下事件名称到相应代码位置 -

一、离线统计
每触发点亮＋1，如果相临两次点亮的时间间隔小于1秒，则记为同一个事件（黄玉辉）==
 
-------------7.27新增------------
每分享一次状态+1
点击开关一次+1
每点击绑定账号一次+1
每点击完成一次+1

///-----------------------------------------------------------------------------
/// 已验收
///-----------------------------------------------------------------------------
已完成打点 每启动应用＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每点击底部直播按钮＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每关注、热门和最新页面展示一次＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每点击主界面右上角的搜索图标进入搜索页面＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每在直播间关注主播＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每点击礼物盒＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每点击礼物盒的发送按钮＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每送出礼物成功/失败＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每点击发送评论按钮＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每点击分享图标＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每展示登录页面＋1（林思敏）【验收：黄玉辉】
已完成打点 每选择一种登录方式＋1（林思敏）【验收：黄玉辉】
已完成打点 每进入直播编辑页面＋1（陈番顺）【验收：黄玉辉】
已完成打点 每关闭直播编辑页面＋1（陈番顺）【验收：黄玉辉】
已完成打点 每点击搜索页面的搜索按钮＋1（李世杰）【验收：黄玉辉】
已完成打点 每点击搜索结果的条目＋1（李世杰）【验收：黄玉辉】
已完成打点 每通过搜索结果页面关注主播＋1（李世杰）【验收：黄玉辉】
已完成打点 每选择分享弹窗中的任何一项＋1（李世杰）【验收：黄玉辉】
已完成打点 每主播进入直播结束页面（陈番顺）【验收：黄玉辉】
已完成打点 每点击直播编辑页面的开始直播按钮＋1（陈番顺）【验收：黄玉辉】
已完成打点 每进入直播间＋1（陈番顺）【验收：黄玉辉】
已完成打点 每用户展示直播结束页面＋1（陈番顺）【验收：黄玉辉】
已完成打点 每点击直播结束页面关注按钮＋1（陈番顺）【验收：黄玉辉】
已完成打点 每展示一条通知+1（陈番顺）【验收：黄玉辉】
已完成打点 每点击一条通知+1（陈番顺）【验收：黄玉辉】
已完成打点 每展示充值页面＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每次点击充值+1（黄玉辉）【验收：黄玉辉】
已完成打点 new页面前四的tag每点击一次+1（林思敏）
已完成打点 每点击一次全球热门话题+1（林思敏）
已完成打点 全球热门话题页面的tag每点击一次tag+1（林思敏）
已完成打点 每点击快速发言+1（林思敏）
已完成打点 每点击完成一次+1（林思敏）
已完成打点 每点击绑定账号一次+1（林思敏）
 
二、实时统计
每进入直播间＋1（陈番顺）【未开始】
每在直播间成功成功点亮＋1（黄玉辉）【未开始】
每在直播间成功进行一次弹幕＋1（黄玉辉）【未开始】


-------------7.27新增------------
主播钻石数变化时上报(钻石数变少时上报)（黄玉辉）【未开始】
 
///-----------------------------------------------------------------------------
/// 已验收
///-----------------------------------------------------------------------------
已完成打点 每成功退出直播间＋1（陈番顺）【验收：黄玉辉】
已完成打点 每在手机屏幕上完全展示直播卡片＋1（其中，快速滑动时，则记录停止后完全展示的直播卡片）（黄玉辉）【验收：黄玉辉】
已完成打点 每点击直播卡片＋1（黄玉辉）【验收：黄玉辉】
已完成打点 每成功分享直播＋1（黄玉辉）【验收：黄玉辉】

每在直播间成功关注主播＋1（黄玉辉）【参考离线统计：每在直播间关注主播＋1】
每在直播间成功进行普通评论＋1（黄玉辉）【参考离线统计：每点击发送评论按钮＋1】
每成功送出一份礼物＋1（黄玉辉）【参考离线统计：每送出礼物成功/失败＋1】
 
*/

/**
 *  @author 黄玉辉
 *  @author 李世杰
 *
 *  @brief 打点统计管理器
 */
@interface FBStatisticsManager : NSObject

/** 上报数据 */
+ (void)report:(id<GeneratedMessageProtocol>)protodata;


//事件行为参数
+ (EventParameter *)eventParameterWithKey:(NSString *)key
                                    value:(NSString *)value;

/**
 *  @brief 事件行为定义
 *
 *  @param sessionID 进入页面时间戳
 *  @param ID        事件ID
 *  @param array     事件行为参数数组
 *  @return 事件
 */
+ (Event *)eventWithSessionId:(SInt64)sessionID
                           ID:(NSString *)ID
         eventParametersArray:(NSArray *)array;

/**
 *  @brief 事件打点
 *
 *  @param array 事件数组
 *
 *  @return 事件打点数据
 */
+ (EventsData *)eventsDataWithEventsArray:(NSArray *)array;

/**
 *  @brief 安装、活跃、主动活跃打点
 *
 *  @return 安装、活跃、主动活跃打点数据
 */
+ (AppActiveData *)appActiveData;

// 直播间事件行为参数
+ (RoomEventParameter *)roomEventParameterWithKey:(NSString *)key
                                            value:(NSString *)value;

/**
 *  @brief 直播间事件
 *
 *  @param moudleId    模块Id
 *  @param positionId  位置id
 *  @param roomId      房间ID
 *  @param broadcastId 主播ID
 *  @param Id          直播间事件行为Id
 *  @param array       直播间事件行为参数列表
 *
 *  @return 直播间事件打点
 */
+ (RoomEvent *)roomEventWithMoudleId:(SInt32)moudleId
                          positionId:(SInt32)positionId
                              roomId:(SInt32)roomId
                         broadcastId:(SInt32)broadcastId
                                  Id:(NSString *)Id
            roomeventParametersArray:(NSArray *)array;

/**
 *  @brief 用户直播间行为数据
 *
 *  @param sessionId 进入页面时间戳
 *  @param array     事件数组
 *
 *  @return 用户直播间行为数据
 */
+ (UserActionData *)userActionDataWithSessionId:(SInt64)sessionId
                                    eventsArray:(NSArray *)array;

//网络类型
+ (int32_t)networkType;

/** 登录状态 */
+ (NSInteger)loginStatus;

@end
